envsec/pkg/awsfed/awsfed.go

```diff
@@ -99,6 +100,48 @@ func (a *AWSFed) AWSCreds(
 	return output.Credentials, nil
 }
 
+// AWSCredsFromIdToken behaves similar to AWSCreds but it takes JWT from input
+// rather than reading from a file or cache. This is to allow web services use
+// this package without having to write every user's JWT in a cache or a file.
+func (a *AWSFed) AWSCredsFromIdToken(
```

Comment:
This is copy pasting much of `AWSCreds`. You can replace the middle portion of that function.

Also suggested naming:

```
AWSCreds
AWSCredsWithLocalCache
```

Reply:
Also I think the can be lower case? Not sure anything outside this package is using them